app/mailboxes/application_mailbox.rb

```diff
@@ -21,17 +21,41 @@ def self.reply_mail?
   def self.support_mail?
     proc do |inbound_mail_obj|
       is_a_support_email = false
-      inbound_mail_obj.mail.to&.each do |email|
-        channel = Channel::Email.find_by('lower(email) = ? OR lower(forward_to_email) = ?', email.downcase, email.downcase)
-        if channel.present?
-          is_a_support_email = true
-          break
-        end
-      end
+
+      is_a_support_email = true if reply_to_mail?(inbound_mail_obj, is_a_support_email)
+      is_a_support_email = true if reply_cc_mail?(inbound_mail_obj, is_a_support_email)
+
       is_a_support_email
     end
   end
 
+  def self.reply_to_mail?(inbound_mail_obj, is_a_support_email)
+    return if is_a_support_email
+
+    inbound_mail_obj.mail.to&.each do |email|
+      channel = Channel::Email.find_by('lower(email) = ? OR lower(forward_to_email) = ?', email.downcase, email.downcase)
+      if channel.present?
+        is_a_support_email = true
+        break
+      end
+    end
+    is_a_support_email
+  end
+
+  def self.reply_cc_mail?(inbound_mail_obj, is_a_support_email)
+    return if is_a_support_email
+
+    inbound_mail_obj.mail.cc&.each do |email|
+      channel = Channel::Email.find_by('lower(email) = ? OR lower(forward_to_email) = ?', email.downcase, email.downcase)
+      if channel.present?
+        is_a_support_email = true
+        break
+      end
+    end
+
+    is_a_support_email
+  end
```

Comment:
can we have a single method instead of both these methods, which accepts the argument of an email array instead of inbound_mail_obj?  Since the logic seems to be repetitive